src/Handshake/PermessageDeflateOptions.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+namespace Ratchet\RFC6455\Handshake;
+
+use Psr\Http\Message\MessageInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+final class PermessageDeflateOptions
+{
+    const MAX_WINDOW_BITS = 15;
+    /* this is a private instead of const for 5.4 compatibility */
+    private static $VALID_BITS = ['8', '9', '10', '11', '12', '13', '14', '15'];
+
+    private $deflateEnabled = false;
+
+    private $server_no_context_takeover;
+    private $client_no_context_takeover;
+    private $server_max_window_bits;
+    private $client_max_window_bits;
+
+    private function __construct() { }
+
+    public static function createDefault() {
+        $new                             = new static();
+        $new->deflateEnabled             = true;
+        $new->client_max_window_bits     = self::MAX_WINDOW_BITS;
+        $new->client_no_context_takeover = false;
+        $new->server_max_window_bits     = self::MAX_WINDOW_BITS;
+        $new->server_no_context_takeover = false;
```

Comment:
Is there any specific reason why snake case and camel case is mixed? From a code style perspective, these properties should be camel case.

Reply:
I just kept the names exactly as-is from the RFC instead of cameling them.